frontend/catalyst/qfunc.py

```diff
@@ -187,26 +184,10 @@ def dynamic_one_shot_partial(
                         f"Native mid-circuit measurement mode does not support {type(m).__name__} "
                         "measurements."
                     )
-            _ = kwargs.get("device", None)
-
-            if not tape.shots:
-                raise qml.QuantumFunctionError(
-                    "dynamic_one_shot is only supported with finite shots."
-                )
```

Comment:
Was this removed because Catalyst gets shots from the device and not the tape? In that case we should validate somewhere that the device has shots.

Reply:
Yes, will validate this.